qiskit/pulse/instructions/instruction.py

```diff
@@ -109,7 +110,7 @@ def id(self) -> int:  # pylint: disable=invalid-name
     @property
     def operands(self) -> Tuple:
         """Return instruction operands."""
-        return self._operands
+        return tuple(map(_format_value, self._operands))
```

Comment:
Could this be called earlier, for performance? I believe operands can only be set upon initialization and updated via `assign_parameters`

Reply:
I think think we need to format this on the fly because in the calibration module we directly create parameterized schedule though it's not necessary to compare two schedules so far. If `assign_parameters` is immutable, we can take that approach. For performance point of view maybe we can use [LRU cache](https://docs.python.org/3/library/functools.html#functools.lru_cache).